common/models/acl.js

```diff
@@ -456,15 +489,34 @@ module.exports = function(ACL) {
         if (err) return callback(err, null);
 
         var resolved = self.resolvePermission(effectiveACLs, req);
+        var permission = resolved && resolved.permission;
         if (resolved && resolved.permission === ACL.DEFAULT) {
-          resolved.permission = (model && model.settings.defaultPermission) || ACL.ALLOW;
+          permission = defaultPermission || ACL.ALLOW;
         }
         debug('---Resolved---');
         resolved.debug();
+
+        // set authorizedRoles in remotingContext options argument if
+        // checkAccessForContext resolved permission status is ALLOW,
+        // else set it to empty object
+        authorizedRoles = permission === ACL.ALLOW ? authorizedRoles : {};
+        setAuthorizedRoles(authorizedRoles);
         return callback(null, resolved);
       });
     });
     return callback.promise;
+
+    // helpers
+    function setAuthorizedRoles(authorizedRoles) {
+      var remotingContext = context.remotingContext;
+      const options = remotingContext && remotingContext.args.options;
+      // authorizedRoles key/value map is added to the options argument only if
+      // the latter exists and is an object. That means that the feature's availability
+      // will depend on the app configuration
+      if (options && typeof options === 'object') { // null is object too
+        options.authorizedRoles = authorizedRoles;
+      }
+    }
```

Comment:
the related helper to perform check ups on context.remotingContext.args.options before proceeding